tasks/cephfs/vstart_runner.py

```
@@ -0,0 +1,782 @@
+            return True
+        else:
+            return False
+
+    def kill(self):
+        log.info("kill ")
+        if self.subproc.pid and not self.finished:
+            log.info("kill: killing pid {0} ({1})".format(
+                self.subproc.pid, self.args))
+            safe_kill(self.subproc.pid)
+        else:
+            log.info("kill: already terminated ({0})".format(self.args))
+
+    @property
+    def stdin(self):
+        class FakeStdIn(object):
+            def __init__(self, mount_daemon):
+                self.mount_daemon = mount_daemon
+
+            def close(self):
+                self.mount_daemon.kill()
+
+        return FakeStdIn(self)
+
+
+class LocalRemote(object):
+    """
+    Amusingly named class to present the teuthology RemoteProcess interface when we are really
+    running things locally for vstart
+
+    Run this inside your src/ dir!
+    """
+
+    def __init__(self):
+        self.name = "local"
+        self.hostname = "localhost"
+
+    def run(self, args, check_status=True, wait=True, stdout=None, stderr=None, cwd=None, stdin=None):
+        # We don't need no stinkin' sudo
+        args = [a for a in args if a != "sudo"]
+
+        for arg in args:
+            if not isinstance(arg, basestring):
+                raise RuntimeError("Oops, can't handle arg {0} type {1}".format(
+                    arg, arg.__class__
+                ))
+
+        log.info("Running {0}".format(args))
+        subproc = subprocess.Popen(args,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   stdin=subprocess.PIPE,
+                                   cwd=cwd)
+
+        if stdin and isinstance(stdin, basestring):
+            # Hack: writing to stdin is not deadlock-safe, but it "always" works
+            # as long as the input buffer is "small"
+            subproc.stdin.write(stdin)
```

Hmm, it looks like this just throws away the stdin param if it's not a string? I'm not sure we even expect it to be a string here, but maybe I've missed something, and surely we shouldn't silently ignore it.